bin/cli.js

```diff
@@ -48,26 +48,30 @@ const cliFlags = yargs
   .option('config', {
     'describe': 'Path to config file',
     'type': 'string',
+    //equivalent to: cliFlags.config = getConfigFromFile(cliFlags.config);
+    'coerce': getConfigFromFile 
```

Comment:
You are not. I realized that in order to have the "--help" message like lighthouse does then yargs must be able to validate if the url is being supplied anywhere. If the argv object in the check function throws an error (which is the red no url message) it will also show the see help for more message. In order to do so it must check if the url is given by the cli or config object which during that check is only a link. It must have that object data beforehand, hence the coerce. From a users point of view it's still a link, this coerce only takes that argument and attempts to read its content. It's the exact same functionality as the code it replaces